src/components/Tree/utils/getReorderedNodes.ts

```diff
@@ -14,7 +14,7 @@ export const getReorderedNodes = (
 
     return sourceItem
         ? moveItems<TreeFlatListItem>(
-              sameLevelNodes.at(targetItemIndex),
+              sameLevelNodes[targetItemIndex < 0 ? sameLevelNodes.length + targetItemIndex : targetItemIndex],
```

Comment:
maybe change to (sonarCloud issue):
`const index = targetItemIndex < 0 ? sameLevelNodes.length + targetItemIndex : targetItemIndex`
`sameLevelNodes[index]`